### ISSUE TITLE
New windows are displayed below inactive fullscreen windows

### ISSUE THREAD

**Issue by scottjad** [stumpwm/stumpwm#632]
New windows that are in a frame where there exists an inactive fullscreened window will have focus but won't be visible.

Reproduce:

In a single empty frame, launch mpv or vlc and fullscreen the window. Launch xterm in that same frame. It should display fine. When xterm is the active window, launch another xterm in that frame. Instead of the new xterm displaying, the fullscreen window will be raised although the new xterm will have focus.

This change in behavior was introduced recently by this PR by @goose121 
https://github.com/stumpwm/stumpwm/pull/607

I don't know what the proper fix is.

If the PR change can't be made to support the old behavior while still supporting dunst then I'd prefer it be reverted.

**Comment by scottjad** [comment]
@goose121 I tried this and it works for me. Thank you very much for fixing this! I didn't get an email notification about your PR despite being subscribed to this issue, so perhaps a comment is necessary in the future.

I had two thoughts that we don't necessarily need to tackle before merging this, and if you don't have the time or inclination for them no worries.

First, I wonder if [these lines](https://github.com/stumpwm/stumpwm/pull/634/commits/c0aa6898b6d7d9a3f1f5265e5fe6f193f5cbbaf4) should really be in geometry-hints. It seems like geometry-hints should just return the hints and have less side-effects. geometry-hints is only called by maximize-window so potentially these side-effects could be in maximize-window.

Second, in the old behavior and in this, when a new window is displayed, the fullscreen window is displayed immediately below it, and potentially covering windows that it wasn't before the new window was opened. For example, if you're in firefox, and open the new file dialog, the main firefox window will then be hidden by the fullscreen window. I think ideally we'd still see the main firefox window below the firefox dialog and nothing would change about the visibility of the fullscreen window simply by opening another window. Another example is where there are two frames on one head, and a fullscreen window in one of the frames. If there are multiple windows in each frame, and the fullscreen window is not visible, then creating a new window in the frame with the fullscreen window can cause the window in the other frame to be obscured by the fullscreen window.

I've noticed that removing the "these lines" mentioned above entirely appears to fix this second issue, but there could be reasons why they are needed that I haven't noticed yet. 

Anyway, if you have any ideas on how to fix this behavior or inclinations to do it that would be a welcome improvement I think.

**Comment by goose121** [comment]
Apologies about the notification; I presumed it would be sent.

I agree with both of your concerns; as for the first one, I could fairly easily fix it if those lines turn out to be necessary. In terms of whether or not they are, I'm not sure; they were added 11 years ago in this commit https://github.com/stumpwm/stumpwm/commit/3ba1e2d8eb8a52e61620b4283ce37d26e1949672, and I can't seem to find anything that had been removed in the commit which was similar. My personal theory is that perhaps it was meant to ensure that fullscreen windows were always on top of all other windows, but I'm not sure.

As it is, I don't have any ideas on how to fix this behaviour besides removing them completely, but I would prefer to have a better idea of why they might be there before doing so.
